compiler/dyno/lib/util/filesystem.cpp

```diff
@@ -114,5 +131,81 @@ bool fileExists(const char* path) {
   return err == 0;
 }
 
+/*
+ * Find the default tmp directory. Try getting the tmp dir from the ISO/IEC
+ * 9945 env var options first, then P_tmpdir, then "/tmp".
+ */
+std::string getTempDir() {
+  std::vector<std::string> possibleDirsInEnv = {"TMPDIR", "TMP", "TEMP", "TEMPDIR"};
+  for (unsigned int i = 0; i < possibleDirsInEnv.size(); i++) {
+    auto curDir = getenv(possibleDirsInEnv[i].c_str());
+    if (curDir != NULL) {
+      return curDir;
+    }
+  }
+#ifdef P_tmpdir
+  return P_tmpdir;
+#else
+  return "/tmp";
+#endif
+}
+
+std::error_code deleteDir(std::string dirname) {
+  // LLVM 5 added remove_directories
+  return llvm::sys::fs::remove_directories(dirname, false);
+}
+
+std::error_code makeTempDir(std::string dirPrefix, std::string& tmpDirPathOut) {
+  std::string tmpdirprefix = std::string(getTempDir()) + "/" + dirPrefix;
+  std::string tmpdirsuffix = ".deleteme-XXXXXX";
+
+  struct passwd* passwdinfo = getpwuid(geteuid());
+  const char* userid;
+  if (passwdinfo == NULL) {
+    userid = "anon";
+  } else {
+    userid = passwdinfo->pw_name;
+  }
+  char* myuserid = strdup(userid);
+  removeSpacesBackslashesFromString(myuserid);
+  std::string tmpDir = tmpdirprefix + std::string(myuserid) + tmpdirsuffix;
+  char* tmpDirMut = strdup(tmpDir.c_str());
+  auto dirRes = std::string(mkdtemp(tmpDirMut));
```

Comment:
I added the comment. Originally I was looking at making that replacement and found this in the LLVM source above `createUniqueDirectory`

```
 // This is a mkdtemp with a different pattern. We use createUniqueEntity mostly
 // for consistency. We should try using mkdtemp.
 ```

and so I thought it might not be worth the effort to change it, but I think we could